src/connector/src/parser/protobuf/parser.rs

```diff
@@ -252,30 +388,84 @@ pub fn from_protobuf_value(field_desc: &FieldDescriptor, value: &Value) -> Resul
             ScalarImpl::Utf8(enum_symbol.name().into())
         }
         Value::Message(dyn_msg) => {
-            let mut rw_values = Vec::with_capacity(dyn_msg.descriptor().fields().len());
-            // fields is a btree map in descriptor
-            // so it's order is the same as datatype
-            for field_desc in dyn_msg.descriptor().fields() {
-                // missing field
-                if !dyn_msg.has_field(&field_desc)
-                    && field_desc.cardinality() == Cardinality::Required
-                {
-                    let err_msg = format!(
-                        "protobuf parse error.missing required field {:?}",
-                        field_desc
-                    );
-                    return Err(RwError::from(ProtocolError(err_msg)));
+            if dyn_msg.descriptor().full_name() == "google.protobuf.Any" {
+                // Sanity check
+                debug_assert!(
+                    dyn_msg.has_field_by_name("type_url") && dyn_msg.has_field_by_name("value"),
+                    "`type_url` & `value` must exist in fields of `dyn_msg`"
+                );
+
+                // The message is of type `Any`
+                let (type_url, payload) = extract_any_info(dyn_msg);
+
+                let payload_field_desc = dyn_msg.descriptor().get_field_by_name("value").unwrap();
+
+                let Some(ScalarImpl::Bytea(payload)) =
+                    from_protobuf_value(&payload_field_desc, &payload, descriptor_pool)?
+                else {
+                    panic!("Expected ScalarImpl::Bytea for payload");
+                };
+
+                // Get the corresponding schema from the descriptor pool
+                let msg_desc = descriptor_pool
+                    .get_message_by_name(&type_url)
+                    .ok_or_else(|| {
+                        ProtocolError(format!(
+                        "Cannot find message {} in from_protobuf_value.\nDescriptor pool is {:#?}",
+                        type_url, descriptor_pool
+                    ))
+                    })?;
+
+                let f = msg_desc
+                    .clone()
+                    .fields()
+                    .map(|f| f.full_name().to_string())
```

Comment:
cc @tabVersion @fuyufjh

Reply:
> Furthermore, f.name() will not conflict because the field names are distinct.

Clients could deliver other messages of any type like this. In this case we could not distinguish between them using only name.
```protubuf
package test_1;
message AnyValue {
  google.protobuf.Any any_value_1 = 1;
  google.protobuf.Any any_value_2 = 2;
}
```

Reply:
Got it. But I'm still not sure if it is necessary to pin down the actual package of the Any message. It's up to the motivation of introducing Any in protobuf parser. If we just need a field to pass through a dynamic structural message, the additional information from `f.full_name()` is not required. Maybe we should leave the decision to @fuyufjh @tabVersion.

Reply:
By the way, if `f.full_name()` is used, the unit tests with prefix `test_any*` should be modified as well.

Reply:
> By the way, if f.full_name() is used, the unit tests with prefix test_any* should be modified as well.

Yes, we'll see if this is expected after further reviewing.